### ISSUE TITLE
Build scripts are failing

### ISSUE THREAD

**Issue by guy-ingo** [yi-sun/circom-pairing#11]
Hi, 

I'm trying to build the optimalate and verify (using `yarn build:optimalate` in the main folder) and I'm get a problem (the example above is for optimalate):
```
****COMPILING CIRCUIT****
invalid output path
previous errors were found
DONE (0s)
****COMPILING C++ WITNESS GENERATION CODE****
build_optimalate.sh: 30: cd: can't cd to ../../build/optimalate/optimalate_cpp
make: *** No targets specified and no makefile found.  Stop.
DONE (0s)
****VERIFYING WITNESS****
build_optimalate.sh: 37: ./optimalate: not found
DONE (0s)
[ERROR] snarkJS: [Error: ENOENT: no such file or directory, open 'witness.wtns'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: 'witness.wtns'
}
```

I saw that there's no witness file, so I tried compiling the circuit separately. But then another problem appears when trying to run circom directly (using `circom optimalate.circom --O1 --r1cs --sym --c` at circom-pairing/scripts/optimalate):
```
thread 'main' panicked at 'Method call in file type_analysis/src/analyzers/unknown_known_analysis.rs line 216', /home/ubuntu/circom/program_structure/src/utils/environment.rs:192:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Do you know what causes this issue? (I'm using circom 2.0.4. Downgrading it to 2.0.3 did nothing)

**Comment by yi-sun** [comment]
You may have to create a `/build` directory at the top level of the repo.  Let me know if that resolves the issue.

**Comment by nalinbhardwaj** [comment]
@guy-ingo your new error is due to a circom compiler bug, see https://github.com/iden3/circom/pull/57. For now, I'd recommend rolling back to an old version of circom by running `git checkout 47883d9 && cargo build --release && cargo install --path circom` in the circom git repo on your local

**Comment by Dimo99** [comment]
I have tried using the https://github.com/iden3/circom/pull/57 patch. But I get the following errors.

```
******COMPILING CIRCUIT****
error[T20460]: Typing error found
    ┌─ "/home/dimo/code/repos/circom-exercises/circom-pairing/circuits/final_exp.circom":319:20
    │
319 │     component pow2[BITLENGTH]; // pow2[i] = C(in^{2^i})
    │                    ^^^^^^^^^ The length of every array must known during the constraint generation phase**
```

**Comment by nalinbhardwaj** [comment]
@Dimo99 Yeah it is intended for the static analysis code iden3 introduced to fail the circuit - I asked about fixing this behaviour in the iden3 Telegram but have yet to hear back about a response/fix (note the https://github.com/iden3/circom/pull/57 patch only fixes the compiler panic, not the actual behaviour). To fix the compilation you should use a version of circom before the static analysis patch applied (by using `git checkout 47883d9 && cargo build --release && cargo install --path circom` on your circom repo in particular)
